src/nodes/chunk_dispatch/chunk_dispatch.h

```diff
@@ -72,6 +72,16 @@ typedef struct ChunkDispatchState
 	 */
 	ChunkDispatch *dispatch;
 	ResultRelInfo *rri;
+	/*
+	 * Save attributes of targetList to calculate correct attribute no: in
+	 * case tuple has a different order incase of MERGE with INSERT
+	 */
+	List *colnames_list;
+	/*
+	 * Save dropped attributes to be used during execution to map
+	 * attributes numbers between different tuples.
+	 */
+	List *dropped_attrList;
```

Comment:
It shouldnt be necessary to pass down the column list from planner to executor. Is there no other way to deal with this?

Reply:
I need actual physical layout of table to determine the correct values of dimension columns from the received tuple especially when columns order in tuple received from child node of ChunkDispatch node is different from actual physical ordering.